app/controllers/web-payments/google-pay/normalise-google-pay-payload.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+// NPM Dependencies
+const humps = require('humps')
+
+const normaliseCardName = cardName => {
+  switch (cardName) {
+    case 'MASTERCARD':
+      return 'master-card'
+    case 'AMEX':
+      return 'american-express'
+    case 'VISA':
+      return 'visa'
+    default:
+      throw new Error('Unrecognised card brand in google pay payload')
+  }
+}
+const nullable = word => {
+  if (word.length === 0 || !word.trim()) {
+    return null
+  }
+  return word
+}
+
+module.exports = payload => {
+  const paymentInfo = {
+    last_digits_card_number: payload.details.paymentMethodData.info.cardDetails,
+    brand: normaliseCardName(payload.details.paymentMethodData.info.cardNetwork),
+    cardholder_name: nullable(payload.payerName || ''),
+    email: nullable(payload.payerEmail || '')
```

Comment:
where does the `payload` comes from? i ask because `payerName` and `payerEmail` aren't data available in the google payload.

Reply:
They come from the payment request. If you look at the test (in this PR too) you can see. The Google payload in their docs is just `payload.details.paymentMethodData`

Reply:
And does the payment request comes from https://github.com/alphagov/pay-frontend/blob/2eee18a63cc6ffe1ae6d64848446e745ae955927/app/assets/javascripts/browsered/web-payments/google-pay.js#L36?

Reply:
Yep